src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/ThrowExceptionForHRTests.cs

```diff
@@ -0,0 +1,108 @@
+        [Theory]
+        [InlineData(unchecked((int)0x80020006))]
+        [InlineData(unchecked((int)0x80020101))]
+        public void ThrowExceptionForHR_NoErrorInfo_ReturnsValidException(int errorCode)
+        {
+            ClearCurrentIErrorInfo();
+
+            bool calledCatch = false;
+            try
+            {
+                Marshal.ThrowExceptionForHR(errorCode);
+            }
+            catch (Exception ex)
+            {
+                calledCatch = true;
+
+                Assert.IsType<COMException>(ex);
+                Assert.Equal(errorCode, ex.HResult);
+                Assert.Null(ex.InnerException);
+                Assert.Null(ex.HelpLink);
+                Assert.NotEmpty(ex.Message);
+                if (PlatformDetection.IsNetCore)
+                {
+                    Assert.Equal("System.Private.CoreLib", ex.Source);
+                }
+                else
+                {
+                    Assert.Equal("mscorlib", ex.Source);
+                }
+                Assert.Contains(nameof(ThrowExceptionForHR_NoErrorInfo_ReturnsValidException), ex.StackTrace);
+                Assert.Contains(nameof(Marshal.ThrowExceptionForHR), ex.TargetSite.Name);
+            }
+
+            Assert.True(calledCatch);
+        }
+
+        public static IEnumerable<object[]> ThrowExceptionForHR_ErrorInfo_TestData()
+        {
+            yield return new object[] { unchecked((int)0x80020006), IntPtr.Zero };
+            yield return new object[] { unchecked((int)0x80020101), IntPtr.Zero };
+            yield return new object[] { unchecked((int)0x80020006), (IntPtr)(-1) };
+            yield return new object[] { unchecked((int)0x80020101), (IntPtr)(-1) };
+        }
+
+        [Theory]
+        [MemberData(nameof(ThrowExceptionForHR_ErrorInfo_TestData))]
+        public void ThrowExceptionForHR_ErrorInfo_ReturnsValidException(int errorCode, IntPtr errorInfo)
+        {
+            ClearCurrentIErrorInfo();
+
+            bool calledCatch = false;
+            try
+            {
+                Marshal.ThrowExceptionForHR(errorCode, errorInfo);
+            }
+            catch (Exception ex)
+            {
+                calledCatch = true;
+
+                Assert.IsType<COMException>(ex);
+                Assert.Equal(errorCode, ex.HResult);
+                Assert.Null(ex.InnerException);
+                Assert.Null(ex.HelpLink);
+                Assert.NotEmpty(ex.Message);
+                if (PlatformDetection.IsNetCore)
+                {
+                    Assert.Equal("System.Private.CoreLib", ex.Source);
+                }
+                else
+                {
+                    Assert.Equal("mscorlib", ex.Source);
+                }
+                Assert.Contains(nameof(ThrowExceptionForHR_ErrorInfo_ReturnsValidException), ex.StackTrace);
+                Assert.Contains(nameof(Marshal.ThrowExceptionForHR), ex.TargetSite.Name);
+            }
+
+            Assert.True(calledCatch);
```

Comment:
General practice here is to put an `Assert.Fail(...);` below the `throw` statement. Then the bool can be removed entirely.

Reply:
Throughout.

Reply:
Not sure such a method exists
> System/Runtime/InteropServices/Marshal/ThrowExceptionForHRTests.cs(22,24): error CS0117: 'Assert' does not contain a definition for 'Fail' [/Users/hugh/Documents/GitHub/corefx/src/System.Runtime.InteropServices/tests/System.Runtime.InteropServices.Tests.csproj]
System/Runtime/InteropServices/Marshal/ThrowExceptionForHRTests.cs(62,24): error CS0117: 'Assert' does not contain a definition for 'Fail' [/Users/hugh/Documents/GitHub/corefx/src/System.Runtime.InteropServices/tests/System.Runtime.InteropServices.Tests.csproj]

Reply:
Sad. This approach is fine then. Thanks.